### DISCUSSION TITLE
Option to choose zip/html format from the extension menu/button

### DISCUSSION THREAD

**Discussion by eugenesvk** [gildas-lormeau/SingleFile#1498] [Ideas]
**Is your feature request related to a problem? Please describe.**
Now that you have added the option of saving zip files to the main extension (congrats!), I'd like to remove one inconvenience of selecting a format, which currently requires a visit to the extension options page.

**Describe the solution you'd like**
An option to select html/zip file format from the extension menu button/context menu
  - via a separate menu item that would switch the default file format without having to visit the optinons page
  - via optional menu items that always save in a given format regardless of what the default is, so for example you could configure in settings to add 2 menu items to have the following 3 options:
    - save selection
    - save selection in html 
    - save selection in zip (universal)

(same for any other ways to save if the user wants to, would also be nice if the order of these options were configurable so you can either have a whole "zip" section or have "zip" item right after each "normal" item)

**Describe alternatives you've considered (optional)**
Using your SingleFileZ extension to continue saving zip files 

**Additional context (optional)**
You already have the option to save profiles, but then those only apply to the whole page while I often need to save just a selection on a page. Also I'd prefer to be able to have a menu item instead of a sub-menu item for some frequent operations

**Comment by redactedscribe** [comment]
Even more generally, it would be cool if we could customise a profile before saving a page. Being able to would also reveal the file format option among all other options. E.g. Save with profile > Custom... > Choose a profile (always display the one which was customised previously first) and then edit it > OK to begin saving the page. Any changes to the profile would only be temporary and be used for that particular save of a page.

**Comment by eugenesvk** [comment]
Interesting, do you mean that you'd have sub-menus in the extension button/context menu that would have **all the options** that are currently available in the Options page, but then changing them? Or do you mean it would still open a new page with those settings, but just not save it as a permanent profile

**Comment by redactedscribe** [comment]
I'm thinking showing a pop-up of the extension's options. Pick one of the profiles, edit it, then continue on to show the new tab with the webpage, then save the page. Any edits to a profile would only be for that save of the webpage. Next time you perform a custom save, show the settings in the state that you last customised them with an option to reset them to the state of one of your saved profiles.

**Comment by gildas-lormeau** [comment]
I'm sorry, I haven't replied to this topic yet because I want to think about it carefully. @eugenesvk , you can define 2 profiles in the options page for this and they will appear in the context menu. @redactedscribe it's already more or less possible to define a profile on the fly with the sidebar in Firefox and Opera. It will also be available when SingleFile migrates to MV3. I think this is the best solution in term of integration.

**Comment by eugenesvk** [comment]
I've already mentioned why the two profiles doesn't help 

> You already have the option to save profiles, but then those only apply to the whole page while I often need to save just a selection on a page. Also I'd prefer to be able to have a menu item instead of a sub-menu item for some frequent operations

**Comment by redactedscribe** [comment]
> it's already more or less possible to define a profile on the fly with the sidebar in Firefox and Opera.

I haven't noticed this and will have to check it out.
